### DISCUSSION TITLE
Question on using Linear Line of Sight (LOS)

### DISCUSSION THREAD

**Discussion by tornadosurvivor** [pelicanmapping/osgearth#2019] [Support Q&A]
I want to check whether there is a direct line of sight from point A to point B, so I am trying to use a LinearLineOfSightNode and call the getHasLOS() function.

The problem is, even when I keep my start and end points constant, the result varies depending on how zoomed in my view is. For e.g. at the default view, the result is True. But if I zoom in enough, the result suddenly becomes False. Yet when I zoom back out to the original view, the result remains as False.

I can't figure out why is this so. My earth file only has one set of elevation data loaded in (one level of detail) so the terrain geometry and elevation shouldn't change. It seems as if the elevation data is not loaded until I zoom in to a certain point. Is there a way to ensure that I always query LOS with the greatest level of detail for the elevation data regardless of the viewpoint?

Thanks in advance.

**Comment by gwaldron** [comment]
The `LinearLineOfSightNode` uses a ray to intersect the triangles in the loaded scene graph. It does not intersect the raw elevation data. So the results will become more accurate the more zoomed in you are. osgEarth does not have a facility to intersect source elevation data.

**Comment by osgwraiESC** [comment]
just a thought, if you have added anything else to the mapnode at your end points, you might also do a setTerrainOnly( true ) on your los node,  the default is false which gave me unexpected results (and made for some head scratching) with a model attached to one or both of the end points
